src/lib.rs

```diff
@@ -136,6 +138,14 @@ impl<'a> FMBuilder<'a> {
         self
     }
 
+    /// If `yes`, then different names cannot match the same text value. For example if `$1` binds
+    /// to `a` then `$2` will refuse to match against `a` (though `$1` will continue to match
+    /// against only `a`). Defaults to `false`.
+    pub fn distinct_name_matching(mut self, yes: bool) -> Self {
+        self.options.distinct_name_matching = yes;
```

Comment:
Kind of unrelated to this feature, but in hindsight, this "yes" idiom is a little confusing.

> self.options.distinct_name_matching = yes;

I initially misinterpreted as:

> self.options.distinct_name_matching = true;

Because in my head, "yes" and "true" are affirmative, and "no" and "false" are not.

I'm not sure you will agree, but I think `val` would have been a better name for `yes`.

Reply:
It's taken from the `regex` library. All the alternatives suck too IMHO.

Reply:
I won't die on this hill, but what's up with `val`?

Reply:
Honestly, I don't really want to think about this: the precedent from `regex` is good enough for me, and it's not part of this PR anyway.

Reply:
Fair enough.

*backs off down hill* :)

Reply:
:)